models/base_model.py

```diff
@@ -20,15 +21,20 @@ def __init__(self, *args, **kwargs):
         """
         Intialize object class with 2 arguments/parameters
         """
-        if len(args) > 0:
-            for key in args[0]:
-                setattr(self, key, args[0][key])
-        else:
-            self.id = str(uuid.uuid4())
+        if (len(kwargs) == 0):
+            self.id = str(uuid4())
             self.created_at = datetime.now()
             self.updated_at = datetime.now()
-        for key in kwargs:
-            print("kwargs: {}: {}".format(key, kwargs[key]))
+            models.storage.new(self)
+        else:
+            for key, value in kwargs.items():
+                if key == "update_at":
+                    self.updated_at = datetime.strptime(value, "%Y-%m-%dT%H:%M:%S.%f")
+                elif key == "created_at":
+                    self.created_at = datetime.strptime(value, "%Y-%m-%dT%H:%M:%S.%f")
+                elif key != "__class__":
+                    setattr(self, key, value)
```

Comment:
This is the correct solution and this code works like a charm, I actually used a similar code before your solution as I needed for task 5, but this is great. Just remove the storage part so we can merge pull request and then merge pull request from task 5